### ISSUE TITLE
Bug when running on windows

### ISSUE THREAD

**Issue by acherrera** [jonathan-s/pelican-obsidian#7]
When running on windows, there is an issue with the formatting and how it interacts with pelican formatting. The first indication of this issue was the URLS were showing up like this: `http://127.0.0.1:8000/%7Bfilename%7D/reference/Databricks/Databricks Tables.md`

In the section of code where the paths are assigned and built: https://github.com/jonathan-s/pelican-obsidian/blob/0180f425355bfdce2fe4ab4b5a658ed95dd23565/pelican/plugins/obsidian/obsidian.py#L55-L60

I added some debug line to see what was going on. 
```
Path: \reference\Databricks/
Filename: Databricks Tables
Fullpath: \reference\Databricks/Databricks Tables
Replaced Link:[Databricks Tables]({filename}\reference\Databricks/Databricks Tables.md)
```

I suspected that the two slashes - back and forward slash - were causing issues. After going in and manually putting links in markdown files it was discovered that ANY backslashes do not work correctly.

Doesn't work: `[Databricks Tables]({filename}\reference\Databricks\Databricks Tables.md)`
Doesn't work: `[Databricks Tables]({filename}\reference\Databricks/Databricks Tables.md)`
Does work: `[Databricks Tables]({filename}/reference/Databricks/Databricks Tables.md)`

Because of this behavior, I think that this issue is actually with how Pelican parses links - it jsut doesn't work with Windows style links

Solution:
Replace all separators with forward slashes regardless of operating system:
```
link_structure = link_structure.replace(os.sep, '/')
```

I plan on making a PR for this soon, but this is a temporary work around until then.

**Comment by jonathan-s** [comment]
Thank you, since I'm not running windows, I won't be able to check this. Happy to accept a PR for it though.

**Comment by jonathan-s** [comment]
Though rather than doing `str.replace`, it might be better to use `pathlib.Path` to build up the path first and then get the string version of that.

**Comment by arulrajnet** [comment]
@acherrera FYI I have solved this by using vscode + [devcontainer](https://github.com/arulrajnet/blog.arulraj.net/tree/master/.devcontainer) in my Windows machine for now.

**Comment by arulrajnet** [comment]
@acherrera This has been fixed in this PR #13. 

Could you check and close the same?
